Description: Fixes the compatibility with the version of maven-dependency-tree in Debian
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java
+++ b/src/main/java/org/apache/maven/plugins/shade/mojo/ShadeMojo.java
@@ -1138,7 +1138,9 @@
                                          List<Dependency> transitiveDeps )
                                              throws DependencyGraphBuilderException
     {
-        DependencyNode node = dependencyGraphBuilder.buildDependencyGraph( project, null );
+        ProjectBuildingRequest pbr = new DefaultProjectBuildingRequest( session.getProjectBuildingRequest() );
+        pbr.setProject( project );
+        DependencyNode node = dependencyGraphBuilder.buildDependencyGraph( pbr, null );
         boolean modified = false;
         for ( DependencyNode n2 : node.getChildren() )
         {
